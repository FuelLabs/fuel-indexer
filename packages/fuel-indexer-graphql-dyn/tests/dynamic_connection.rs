pub use async_graphql::dynamic::Schema;
pub use async_graphql::Request;
use fuel_indexer_graphql_dyn::testing::prelude::*;
use insta::*;

#[tokio::test]
async fn test() {
    let schema = Schema::build_test();
    let schema = schema.finish().unwrap();

    // Execute a query
    let response = schema
        .execute(Request::new(
            r#"
            query {
                chain(id: "Chain:0") {
                  a: transactions(first: 1) {
                      edges {
                          node {
                              id
                          }
                          cursor
                      }
                      pageInfo {
                          hasNextPage
                          hasPreviousPage
                          startCursor
                          endCursor
                      }
                  }
                  b: transactions(first: 1) {
                      edges {
                          node {
                              id
                          }
                          cursor
                      }
                      pageInfo {
                          hasNextPage
                          hasPreviousPage
                          startCursor
                          endCursor
                      }
                  }
                }
            }
        "#,
        ))
        .await
        .into_result()
        .unwrap();

    assert_json_snapshot!(response.data);
}
