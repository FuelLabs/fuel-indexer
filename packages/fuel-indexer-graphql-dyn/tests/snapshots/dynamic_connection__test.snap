---
source: packages/fuel-indexer-graphql-dyn/tests/dynamic_connection.rs
expression: response.data
---
{
  "chain": {
    "a": {
      "edges": [
        {
          "node": {
            "id": "Transaction:0-0-0"
          },
          "cursor": "2"
        }
      ],
      "pageInfo": {
        "hasNextPage": false,
        "hasPreviousPage": false,
        "startCursor": "2",
        "endCursor": "2"
      }
    },
    "b": {
      "edges": [
        {
          "node": {
            "id": "Transaction:0-0-0"
          },
          "cursor": "2"
        }
      ],
      "pageInfo": {
        "hasNextPage": false,
        "hasPreviousPage": false,
        "startCursor": "2",
        "endCursor": "2"
      }
    }
  }
}
